src/internal/methods/NeoEspBitBangMethod.cpp

```diff
@@ -137,19 +149,44 @@ void IRAM_ATTR neoEspBitBangWriteSpacingPixels(const uint8_t* pixels,
             subpix = *pixels++;
 
             // if pixel spacing is needed
-            if (tSpacing)
+            element++;
+            if (element == sizePixel)
             {
-                element++;
-                if (element == sizePixel)
+                element = 0;
+
+                // Hack: permit interrupts to fire
+                // If we get held up more than the latch period, restart.
+                // We do this at the end of an element to ensure that each element always gets *some* valid 
+#if !defined(ARDUINO_ARCH_ESP32)
+                interrupts();
+#endif                
+                // wait for pixel spacing
+                if (tSpacing)
                 {
-                    element = 0;
-
-                    // wait for pixel spacing
                     while ((getCycleCount() - cyclesNext) < tSpacing);
                 }
+#if !defined(ARDUINO_ARCH_ESP32)
+                noInterrupts();
+#endif
+                if ((getCycleCount() - cyclesNext) > tReset) {
+                    if (++retries > 8) break;   // give up
+                    pixels = p_start;
+                    subpix = *pixels++;
+                    mask = 0x80;
+                    element = 0;
+                }
             }
         }
     }
+
+        
+#if defined(ARDUINO_ARCH_ESP32)
+    portEXIT_CRITICAL(&updateMux);
+#else
+    interrupts();
+#endif
+
+    if (retries > 0) Serial.printf("NPB retries: %d\n",retries);
```

Comment:
No Serial access from within the core unless it's wrapped into debugging only.  Either comment this out or wrap with an appropriate system log function (does esp8266 even have these?)

Reply:
Sorry I missed that - debugging code I should've eliminated.  Thanks for the catch.